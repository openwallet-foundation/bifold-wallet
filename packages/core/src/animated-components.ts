import ButtonLoading from './components/animated/ButtonLoading'
import ConnectionLoading from './components/animated/ConnectionLoading'
import CredentialAdded from './components/animated/CredentialAdded'
import CredentialPending from './components/animated/CredentialPending'
import LoadingIndicator from './components/animated/LoadingIndicator'
import RecordLoading from './components/animated/RecordLoading'
import SendingProof from './components/animated/SendingProof'
import SentProof from './components/animated/SentProof'

export interface AnimatedComponents {
  ButtonLoading: React.FC
  ConnectionLoading: React.FC
  CredentialAdded: React.FC
  CredentialPending: React.FC
  LoadingIndicator: React.FC
  RecordLoading: React.FC
  SendingProof: React.FC
  SentProof: React.FC
}

export const animatedComponents: AnimatedComponents = {
  ButtonLoading: ButtonLoading,
  ConnectionLoading: ConnectionLoading,
  CredentialAdded: CredentialAdded,
  CredentialPending: CredentialPending,
  LoadingIndicator: LoadingIndicator,
  RecordLoading: RecordLoading,
  SendingProof: SendingProof,
  SentProof: SentProof,
}
