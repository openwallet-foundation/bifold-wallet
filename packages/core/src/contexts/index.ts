import * as reducers from './reducers'
import * as store from './store'
export { reducers, store }
