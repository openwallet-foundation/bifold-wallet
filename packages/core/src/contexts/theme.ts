import { createContext, useContext } from 'react'

import { theme, ITheme } from '../theme'

export const ThemeContext = createContext<ITheme>(theme)

export const ThemeProvider = ThemeContext.Provider

export const useTheme = () => useContext(ThemeContext)
