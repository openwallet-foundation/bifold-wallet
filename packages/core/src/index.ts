/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@credo-ts/core'
import AgentProvider from '@credo-ts/react-hooks'

import createApp from './App'
import * as components from './components'
import { ButtonImpl as Button, ButtonType, Button as IButton } from './components/buttons/Button'
import IconButton, { ButtonLocation } from './components/buttons/IconButton'
import BulletPoint from './components/inputs/BulletPoint'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import LimitedTextInput from './components/inputs/LimitedTextInput'
import NotificationListItem from './components/listItems/NotificationListItem'
import ContentGradient from './components/misc/ContentGradient'
import CredentialCard from './components/misc/CredentialCard'
import ErrorBoundaryWrapper from './components/misc/ErrorBoundary'
import FauxHeader from './components/misc/FauxHeader'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import QRScannerTorch from './components/misc/QRScannerTorch'
import DeveloperModal from './components/modals/DeveloperModal'
import ErrorModal from './components/modals/ErrorModal'
import SafeAreaModal from './components/modals/SafeAreaModal'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import Link from './components/texts/Link'
import Text from './components/texts/Text'
import { ThemedText } from './components/texts/ThemedText'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { credentialOfferTourSteps } from './components/tour/CredentialOfferTourSteps'
import { credentialsTourSteps } from './components/tour/CredentialsTourSteps'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { proofRequestTourSteps } from './components/tour/ProofRequestTourSteps'
import { TourBox } from './components/tour/TourBox'
import { Banner, BannerMessage, BannerSection } from './components/views/Banner'
import HomeFooterView from './components/views/HomeFooterView'
import KeyboardView from './components/views/KeyboardView'
import { attemptLockoutConfig, PINRules, tours, walletTimeout } from './constants'
import { defaultConfig, defaultHistoryEventsLogger } from './container-impl'
import * as contexts from './contexts'
import { ActivityProvider, AutoLockTime, useActivity } from './contexts/activity'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import { useDeveloperMode } from './hooks/developer-mode'
import usePreventScreenCapture from './hooks/screen-capture'
import useBifoldAgentSetup from './hooks/useBifoldAgentSetup'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { DefaultScreenLayoutOptions } from './navigators/defaultLayoutOptions'
import { DefaultScreenOptionsDictionary, useDefaultStackOptions } from './navigators/defaultStackOptions'
import AttemptLockout from './screens/AttemptLockout'
import Biometry from './screens/Biometry'
import Developer from './screens/Developer'
import Onboarding from './screens/Onboarding'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Scan from './screens/Scan'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UpdateAvailable from './screens/UpdateAvailable'
import { bifoldLoggerInstance } from './services/bifoldLogger'
import { isBiometricsActive, loadLoginAttempt } from './services/keychain'
import { BifoldLogger } from './services/logger'
import { AbstractBifoldLogger } from './services/AbstractBifoldLogger'
import { DeepPartial, ThemeBuilder } from './theme-builder'
import * as types from './types'
import { CredentialListFooterProps } from './types/credential-list-footer'

export { animatedComponents } from './animated-components'
export { EventTypes, LocalStorageKeys } from './constants'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { useAuth } from './contexts/auth'
export { useNetwork } from './contexts/network'
export { DispatchAction, reducer, default as Store } from './contexts/reducers/store'
export { defaultState, mergeReducers, StoreContext, StoreProvider, useStore } from './contexts/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { useDeepLinks } from './hooks/deep-links'
export { initLanguages, initStoredLanguage, Locales, translationResources } from './localization'
export * from './navigators'
export { createStyles } from './screens/OnboardingPages'
export * from './services/storage'
export { bifoldTheme, ColorPalette, Assets as ImageAssets } from './theme'
export * from './types/attestation'
export { BifoldError } from './types/error'
export { Screens, Stacks, TabStacks } from './types/navigators'
export * from './types/version-check'
export { createLinkSecretIfRequired, getAgentModules } from './utils/agent'
export { getCredentialIdentifiers, isValidAnonCredsCredential } from './utils/credential'
export {
  connectFromScanOrDeepLink,
  formatTime,
  getConnectionName,
  removeExistingInvitationsById,
  useCredentialConnectionLabel,
} from './utils/helpers'
export { getIndyLedgers, IndyLedger, readIndyLedgersFromFile, writeIndyLedgersToFile } from './utils/ledger'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { migrateToAskar } from './utils/migration'
export { buildFieldsFromAnonCredsCredential } from './utils/oca'
export { testIdForAccessabilityLabel, testIdWithKey } from './utils/testable'

export type { AnimatedComponents } from './animated-components'
export type { ReducerAction } from './contexts/reducers/store'
export type { RenderProps, TourStep } from './contexts/tour/tour-context'
export type {
  IAssets,
  IBrandColors,
  IColorPalette,
  IErrorColors,
  IFontAttributes,
  IGrayscaleColors,
  IInlineInputMessage,
  IInputAttributes,
  IInputs,
  INotificationColors,
  ISemanticColors,
  ISpacing,
  ISVGAssets,
  ITabTheme,
  ITextTheme,
  ITheme,
} from './theme'
export type { GenericFn } from './types/fn'
export { BasicMessageMetadata, CredentialMetadata } from './types/metadata'
export type { basicMessageCustomMetadata, credentialCustomMetadata } from './types/metadata'
export type { ContactStackParams, NotificationStackParams, OnboardingStackParams } from './types/navigators'
export type { WalletSecret } from './types/security'
export type {
  LoginAttempt as LoginAttemptState,
  Migration as MigrationState,
  Onboarding as OnboardingState,
  PersistentState,
  Preferences as PreferencesState,
  State,
  Tours as ToursState,
  VersionInfo,
} from './types/state'
export type { BifoldAgent } from './utils/agent'
export type { IndyLedgerConfig, IndyLedgerFileSystem, IndyLedgerJSON, IndyLedgersRecord } from './utils/ledger'
export type { OnboardingStyleSheet }

export type { InlineMessageProps } from './components/inputs/InlineErrorText'

export { InlineErrorPosition } from './types/error'

export * from './container-api'
export { MainContainer } from './container-impl'
export type { CredentialListFooterProps }

export type { Config, HistoryEventsLoggerConfig } from './types/config'
export type { OnboardingTask, ScreenLayoutConfig } from './types/navigators'

export { LockoutReason } from './contexts/auth'
export type { OnboardingStackProps } from './navigators/OnboardingStack'
export type { SplashProps } from './screens/Splash'
export { BaseTourID } from './types/tour'

export type { BannerSectionProps } from './components/views/Banner'

export {
  ActivityProvider,
  Agent,
  AgentProvider,
  AttachTourStep,
  AttemptLockout,
  attemptLockoutConfig,
  AuthProvider,
  AutoLockTime,
  Banner,
  BannerSection,
  AbstractBifoldLogger,
  BifoldLogger,
  bifoldLoggerInstance,
  Biometry,
  BulletPoint,
  Button,
  ButtonLocation,
  ButtonType,
  CheckBoxRow,
  components,
  ContentGradient,
  contexts,
  createApp,
  CredentialCard,
  credentialOfferTourSteps,
  credentialsTourSteps,
  defaultConfig,
  defaultHistoryEventsLogger,
  DefaultScreenLayoutOptions,
  DefaultScreenOptionsDictionary,
  Developer,
  DeveloperModal,
  ErrorBoundaryWrapper,
  ErrorModal,
  FauxHeader,
  HomeFooterView as HomeContentView,
  homeTourSteps,
  IconButton,
  InfoBox,
  InfoBoxType,
  InfoTextBox,
  isBiometricsActive,
  KeyboardView,
  LimitedTextInput,
  Link,
  loadLoginAttempt,
  NavContainer,
  NetworkProvider,
  NotificationListItem,
  Onboarding,
  OnboardingPages,
  OpenIDCredentialRecordProvider,
  PINRules,
  Preface,
  proofRequestTourSteps,
  QRScannerTorch,
  Record,
  SafeAreaModal,
  Scan,
  Splash,
  Terms,
  Text,
  ThemeBuilder,
  ThemedText,
  toastConfig,
  ToastType,
  TourBox,
  TourProvider,
  tours,
  types,
  UpdateAvailable,
  useActivity,
  useBifoldAgentSetup,
  useDefaultStackOptions,
  useDeveloperMode,
  usePreventScreenCapture,
  useTour,
  walletTimeout,
}
export type { BannerMessage, DeepPartial, IButton }
