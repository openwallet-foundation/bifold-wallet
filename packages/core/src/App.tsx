import React, { useEffect } from 'react'
import { StatusBar } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import { useNavigationContainerRef } from '@react-navigation/native'
import { isTablet } from 'react-native-device-info'
import Orientation from 'react-native-orientation-locker'
import { animatedComponents } from './animated-components'
import ErrorModal from './components/modals/ErrorModal'
import toastConfig from './components/toast/ToastConfig'
import { tours } from './constants'
import { Container, ContainerProvider } from './container-api'
import { AnimatedComponentsProvider } from './contexts/animated-components'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { StoreProvider } from './contexts/store'
import { ThemeProvider } from './contexts/theme'
import { TourProvider } from './contexts/tour/tour-provider'
import { initStoredLanguage } from './localization'
import RootStack from './navigators/RootStack'
import { bifoldTheme, themes } from './theme'

const createApp = (container: Container): React.FC => {
  const AppComponent: React.FC = () => {
    const navigationRef = useNavigationContainerRef()

    useEffect(() => {
      initStoredLanguage().then()
    }, [])

    useEffect(() => {
      // Hide the native splash / loading screen so that our
      // RN version can be displayed.
      SplashScreen.hide()
    }, [])

    if (!isTablet()) {
      Orientation.lockToPortrait()
    }

    return (
      <ContainerProvider value={container}>
        <StoreProvider>
          <ThemeProvider themes={themes} defaultThemeName={bifoldTheme.themeName}>
            <NavContainer navigationRef={navigationRef}>
              <AnimatedComponentsProvider value={animatedComponents}>
                <AuthProvider>
                  <NetworkProvider>
                    <StatusBar
                      hidden={false}
                      barStyle="light-content"
                      backgroundColor={bifoldTheme.ColorPalette.brand.primary}
                      translucent={false}
                    />
                    <ErrorModal />
                    <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
                      <RootStack />
                    </TourProvider>
                    <Toast topOffset={15} config={toastConfig} />
                  </NetworkProvider>
                </AuthProvider>
              </AnimatedComponentsProvider>
            </NavContainer>
          </ThemeProvider>
        </StoreProvider>
      </ContainerProvider>
    )
  }

  return AppComponent
}

export default createApp
