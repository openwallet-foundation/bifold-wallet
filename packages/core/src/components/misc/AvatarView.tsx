import { hashCode, hashToRGBA } from '@hyperledger/aries-oca'
import React from 'react'
import { StyleSheet, Text, View, ViewStyle } from 'react-native'

import { useTheme } from '../../contexts/theme'
import { testIdWithKey } from '../../utils/testable'

interface AvatarViewProps {
  name: string
  style?: ViewStyle
}

const AvatarView: React.FC<AvatarViewProps> = ({ name, style }) => {
  const { ListItems } = useTheme()
  const styles = StyleSheet.create({
    avatar: {
      ...ListItems.avatarCircle,
      margin: 12,
      borderWidth: 3,
      justifyContent: 'center',
      alignItems: 'center',
    },
  })
  return (
    <View style={[styles.avatar, { borderColor: hashToRGBA(hashCode(name)) }, style]}>
      <Text style={ListItems.avatarText} testID={testIdWithKey('AvatarName')}>
        {name.charAt(0)}
      </Text>
    </View>
  )
}

export default AvatarView
