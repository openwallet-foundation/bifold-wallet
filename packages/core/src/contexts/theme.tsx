import { createContext, useCallback, useContext, useMemo } from 'react'

import { bifoldTheme, ITheme } from '../theme'
import { useStore } from './store'
import { DispatchAction } from './reducers/store'

export interface IThemeContext extends ITheme {
  setTheme: (themeName: string) => void
}

const ThemeContext = createContext<IThemeContext>({
  ...bifoldTheme,
  setTheme: () => {},
})

export interface ThemeProviderProps extends React.PropsWithChildren {
  themes: ITheme[]
  defaultThemeName: string
}

export const ThemeProvider = ({ themes, defaultThemeName, children }: ThemeProviderProps) => {
  const [store, dispatch] = useStore()

  const activeTheme = useMemo(() => {
    return (
      (store.preferences.theme && themes.find((t) => t.themeName === store.preferences.theme)) ||
      themes.find((t) => t.themeName === defaultThemeName) ||
      themes[0]
    )
  }, [store.preferences.theme, themes, defaultThemeName])

  const setTheme = useCallback(
    (themeName: string) => {
      const newTheme = themes.find((t) => t.themeName === themeName) || themes[0]
      dispatch({ type: DispatchAction.SET_THEME, payload: [newTheme.themeName] })
    },
    [themes, dispatch]
  )

  // prevent re-rendering of the context value
  const value = useMemo(() => {
    return {
      ...activeTheme,
      setTheme,
    }
  }, [activeTheme, setTheme])

  return <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>
}

export const useTheme = () => useContext(ThemeContext)
