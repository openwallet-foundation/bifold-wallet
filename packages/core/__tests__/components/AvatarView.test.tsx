import { render } from '@testing-library/react-native'
import React from 'react'

import AvatarView from '../../src/components/misc/AvatarView'
import * as themeContext from '../../src/contexts/theme' // note we're importing with a * to import all the exports
import { theme } from '../../src/theme'

describe('AvatarView Component', () => {
  test('Renders correctly', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(<AvatarView name={'Bacon'} />)

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly 2', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(<AvatarView name={'Bacon'} />)

    expect(tree).toMatchSnapshot()
  })
})
