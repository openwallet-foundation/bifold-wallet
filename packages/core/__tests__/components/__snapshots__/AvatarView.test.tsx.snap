// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarView Component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 32,
        "borderWidth": 3,
        "height": 64,
        "justifyContent": "center",
        "margin": 12,
        "width": 64,
      },
      Object {
        "borderColor": "#060741",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 32,
        "fontWeight": "normal",
      }
    }
    testID="com.ariesbifold:id/AvatarName"
  >
    B
  </Text>
</View>
`;

exports[`AvatarView Component Renders correctly 2 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 32,
        "borderWidth": 3,
        "height": 64,
        "justifyContent": "center",
        "margin": 12,
        "width": 64,
      },
      Object {
        "borderColor": "#060741",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 32,
        "fontWeight": "normal",
      }
    }
    testID="com.ariesbifold:id/AvatarName"
  >
    B
  </Text>
</View>
`;
